src/sbml/UnitDefinition.cpp

```diff
@@ -2475,13 +2395,13 @@ LIBSBML_EXTERN
 UnitDefinition_t*
 UnitDefinition_clone (const UnitDefinition_t *ud)
 {
-  return (ud != NULL) ? static_cast<UnitDefinition_t*>( ud->clone() ) : NULL;
```

Comment:
Okay these need to be left as essentially it is C code but being compiled by C++
a struct implies public inheritance by default ... class private inheritance by default
if that were to be enforced it could make a difference ... though i think we are always explicit with the inheritance we choose

so need to leave static casts

Reply:
I feel like I might be missing something here; I'm not sure how the inheritance policy changes how this pointer is handled. Here's how I understand it:

If `ud` represents some derived class/struct of `UnitDefinition_t` then the virtual method `ud->clone()` climbs down the inheritance tree and returns a pointer to the derived class. If the `static_cast` isn't there then that pointer is implicitly cast to `UnitDefinition_t*` on return. Unless I've missed something, the function of the static_cast is simply to perform this cast explicitly (though redundantly). Whether `ud` is an instance of a class or a struct doesn't seem to make a different.

Reply:
Agreed in this case it probably doesnt BUT other classes with multiple inheritance it might !

Reply:
Even in the case of multiple inheritance, because the return type of `UnitDefinition_clone` is `UnitDefinition_t*`, the return value of `ud-clone()` will be implicitly cast to the correct type, without the need for a static_cast. I can certainly keep the static_casts in if you feel strongly about it but I'm certain they're unnecessary in these cases.